### PR TITLE
wrap clickhouse write in a retry loop

### PR DESCRIPTION
**Pull request by luluz66**
https://github.com/buildbuddy-io/buildbuddy-internal/issues/1723

### CODE REVIEW

File: server/util/clickhouse/clickhouse.go [hunk 1]
```diff
@@ -175,8 +183,27 @@ func ToInvocationFromPrimaryDB(ti *tables.Invocation) *Invocation {
...
 func (h *DBHandle) FlushInvocationStats(ctx context.Context, ti *tables.Invocation) error {
 	inv := ToInvocationFromPrimaryDB(ti)
-	res := h.DB(ctx).Create(inv)
-	return res.Error
+	retrier := retry.New(ctx, &retry.Options{
+		MaxRetries:     numClickhouseRetries,
+		InitialBackoff: 50 * time.Millisecond,
+		MaxBackoff:     3 * time.Second,
+		Multiplier:     2,
+	})
```
Comment (tylerwilliams) [inline server/util/clickhouse/clickhouse.go:191]: I'm not sure this buys you much over the stock retrier. I would probably just use that one unless you have a compelling reason not to?

Comment (luluz66) [reply server/util/clickhouse/clickhouse.go]: Done.

File: server/util/clickhouse/clickhouse.go [hunk 2]
```diff
@@ -3,20 +3,27 @@ package clickhouse
...
 	"github.com/buildbuddy-io/buildbuddy/server/util/log"
+	"github.com/buildbuddy-io/buildbuddy/server/util/retry"
 	"github.com/buildbuddy-io/buildbuddy/server/util/status"
 	gormclickhouse "gorm.io/driver/clickhouse"
 	"gorm.io/gorm"
 )
 
+const (
+	numClickhouseRetries = 3
+)
```
Comment (tylerwilliams) [inline server/util/clickhouse/clickhouse.go:25]: you can remove this now

**Review by tylerwilliams — APPROVED** [review]

File: server/util/clickhouse/clickhouse.go (hunk 2 above)
Comment (luluz66) [reply server/util/clickhouse/clickhouse.go]: Done

**Pull request merged**